token-metadata/program/src/processor/collection/unverify_collection.rs

```diff
@@ -1,71 +1,122 @@
-use borsh::BorshSerialize;
 use mpl_utils::assert_signer;
 use solana_program::{
     account_info::{next_account_info, AccountInfo},
     entrypoint::ProgramResult,
     pubkey::Pubkey,
+    system_program,
 };
 
 use crate::{
     assertions::{
-        assert_owned_by,
-        collection::{assert_collection_verify_is_valid, assert_has_collection_authority},
+        assert_owned_by, collection::assert_has_collection_authority,
+        metadata::assert_metadata_derivation,
     },
     error::MetadataError,
     state::{Metadata, TokenMetadataAccount},
+    utils::clean_write_metadata,
 };
 
 pub fn unverify_collection(program_id: &Pubkey, accounts: &[AccountInfo]) -> ProgramResult {
     let account_info_iter = &mut accounts.iter();
     let metadata_info = next_account_info(account_info_iter)?;
     let collection_authority_info = next_account_info(account_info_iter)?;
-    let collection_mint = next_account_info(account_info_iter)?;
-    let collection_info = next_account_info(account_info_iter)?;
+    let collection_mint_info = next_account_info(account_info_iter)?;
+    let collection_metadata_info = next_account_info(account_info_iter)?;
     let edition_account_info = next_account_info(account_info_iter)?;
+
     let using_delegated_collection_authority = accounts.len() == 6;
 
-    assert_signer(collection_authority_info)?;
+    // Account validation.
     assert_owned_by(metadata_info, program_id)?;
-    assert_owned_by(collection_info, program_id)?;
-    assert_owned_by(collection_mint, &spl_token::id())?;
-    assert_owned_by(edition_account_info, program_id)?;
+    assert_signer(collection_authority_info)?;
+    assert_owned_by(collection_mint_info, &spl_token::id())?;
```

Comment:
Does the mint account persist even after a full burn? Wouldn't it be returned to the system program?

Reply:
Mints from the original SPL token program cannot be closed. If we add support for SPL_Token_2022 then we would need to change this logic.